masonry/src/render_root.rs

```diff
@@ -489,6 +498,19 @@ impl RenderRoot {
             self.global_state
                 .emit_signal(RenderRootSignal::RequestRedraw);
         }
+
+        if self.global_state.is_ime_active {
+            let widget = self
+                .global_state
+                .focused_widget
+                .expect("IME is active without a focused widget");
+            let ime_area = self.widget_arena.get_state(widget).item.get_ime_area();
```

Comment:
It's conceivable that you have enough context to call a `Widget` method with a `QueryCtx` here.

Reply:
Hmm, that's true, and it's probably too messy. Without having a new invalidation method, it doesn't really work.